### DISCUSSION TITLE
List what `extra_kwargs` exist for a model serializer

### DISCUSSION THREAD

**Discussion by mangelozzi** [encode/django-rest-framework#8877] [Ideas & Suggestions]
[Serializers - [Additional keyword arguments](https://www.django-rest-framework.org/api-guide/serializers/#additional-keyword-arguments)](https://www.django-rest-framework.org/api-guide/serializers/#additional-keyword-arguments) says:

>> this option is a dictionary, mapping field names to a dictionary of keyword arguments. For example:

What keyword arguments are support? From the example I only can find `write_only`.

How is one supposed to figure out it out, e.g. Is `required` supported?, `blank`? Should they not be listed somewhere?

**Comment by lovelydinosaur** [comment]
Which keywords are supported will depend on which serialiser field you're adding them too.

**Reply by mangelozzi** [reply]
> Which keywords are supported will depend on which serialiser field you're adding them too.

Thanks for the reply.

Can they at least all take the core args? These ones listed here: 
https://www.django-rest-framework.org/api-guide/fields/#core-arguments ? If yes it would be handy in the documentation to point this out to people new to using DRF, maybe its obvious to everyone else.

Say I have a `serializers.CharField`, can any of the init args for the constructor can be used as `extra_kwargs`?

**Comment by lovelydinosaur** [comment]
> Can they at least all take the core args?

Yes. Perhaps the "[additional keyword arguments on fields](https://www.django-rest-framework.org/api-guide/serializers/#additional-keyword-arguments)" text could hyperlink to the "[Serializer Fields](https://www.django-rest-framework.org/api-guide/fields/)" documentation page. That might provide enough extra context for users?
